src/praisonai-agents/praisonaiagents/memory/memory.py

```
@@ -422,7 +422,14 @@ def _init_chroma(self):
             os.makedirs(rag_path, exist_ok=True)
 
             # Get chromadb lazily
-            chromadb, ChromaSettings = _get_chromadb()
+            try:
+                import chromadb
+                from chromadb.config import Settings as ChromaSettings
+            except ImportError:
+                raise ImportError(
+                    "chromadb is required for the rag memory provider. "
+                    "Install with: pip install 'praisonaiagents[memory]'"
+                )
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)